Bio 1876 adapt to nf core sarek rb

b97pla opened the pull request:
This PR introduces breaking changes w.r.t. handling the Sarek pipeline. 

The Sarek engine has been rewritten to handle Sarek >= 2.6. Here are the main changes:

- Only the main Sarek workflow is invoked
- Separated config sections for nextflow and sarek 
- Adjust the genome_base_path parameter for GRCh37 and GRCh38
- By default, the resume flag is used
- Added a script to parse results and update Charon independently from the local tracking database
- Added some scripts to build docker and singularity containers (won't be able to run SLURM from the container though)

matrulda commented:
Wow, great work! Just wanted to drop a comment that this review is on my to-do list, other things has gotten in the way. It's quite a big PR, so I'll have to spend some time on it.

matrulda commented on ngi_pipeline/tests/engines/sarek/models/test_sarek.py:
```diff
@@ -102,29 +104,86 @@ def _alignment_status_wrapper(status, *args):
+        for key in config["nextflow"].keys():
+            self.assertEqual(config["nextflow"][key], sarek_analysis.nextflow_config[key])
+
+        del(config["nextflow"]["profile"])
```
This test is really long and a bit hard to follow, I think it could benefit from being divided into different tests. For example this part when testing if the default value is used.

b97pla replied:
good point, I'll look into that

matrulda commented on requirements-dev.txt:
```diff
@@ -1,7 +1,7 @@
 -r requirements.txt
-codecov==2.0.9
+codecov
```
Why do you remove the specific versions? I had problems with running the tests since I had a too old version of PyAML.

b97pla replied:
ok, this was probably a mistake. PyCharm was complaining a lot about old dependencies but I think I also had some problem with incompatible dependencies when creating the docker image and taking away the versions fixed that. I'll go over it again and see if I can fix it.

aanil commented on docker/ngi_config.yaml:
```diff
@@ -0,0 +1,127 @@
+    gatk_key: "/path/to/gatk.key"
+
+sarek:
+    tag: 2.6
```
Sarek 2.6.1 has now been released with GATK updated. Version 2.6 didn't work completely when we tried to run it on Irma.

b97pla replied:
thanks, I'll update this config and also the upcoming PR to irma-provision when this has been merged

b97pla commented on docker/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+FROM continuumio/miniconda:latest
```
do you mean building and deploying the docker image to docker hub pulling from there when building the singularity image? I'm not sure having to maintain that outweighs the hassle of creating the docker image first (given that the use of this image would be very limited).. Or do you mean basically translating the Dockerfile to a Singularity recipe? I guess that could be done, it's just that my experience with singularity recipes is very limited so I stuck with the Dockerfile..

matrulda replied:
I meant translating it to a singularity recipe. The docker image takes up some space. But it's not that important.

b97pla commented:
Thanks for the reviews and apologies for the size of the PR. 

I have restored the versions on the dependencies and fixed the conflicts, as well as refactored the test. I also fixed some of the code complexity that CodeClimate was complaining about.

matrulda reviewed (approved):
Nice, looks good to me!
